adder-codec-core/src/lib.rs

```diff
@@ -346,6 +346,268 @@ pub const D_SHIFT: [UDshift; 128] = [
+    (1_u128 << 24) as f64,
+    (1_u128 << 25) as f64,
+    (1_u128 << 26) as f64,
+    (1_u128 << 27) as f64,
+    (1_u128 << 28) as f64,
+    (1_u128 << 29) as f64,
+    (1_u128 << 30) as f64,
+    (1_u128 << 31) as f64,
+    (1_u128 << 32) as f64,
+    (1_u128 << 33) as f64,
+    (1_u128 << 34) as f64,
+    (1_u128 << 35) as f64,
+    (1_u128 << 36) as f64,
+    (1_u128 << 37) as f64,
+    (1_u128 << 38) as f64,
+    (1_u128 << 39) as f64,
+    (1_u128 << 40) as f64,
+    (1_u128 << 41) as f64,
+    (1_u128 << 42) as f64,
+    (1_u128 << 43) as f64,
+    (1_u128 << 44) as f64,
+    (1_u128 << 45) as f64,
+    (1_u128 << 46) as f64,
+    (1_u128 << 47) as f64,
+    (1_u128 << 48) as f64,
+    (1_u128 << 49) as f64,
+    (1_u128 << 50) as f64,
+    (1_u128 << 51) as f64,
+    (1_u128 << 52) as f64,
+    (1_u128 << 53) as f64,
+    (1_u128 << 54) as f64,
+    (1_u128 << 55) as f64,
+    (1_u128 << 56) as f64,
+    (1_u128 << 57) as f64,
+    (1_u128 << 58) as f64,
+    (1_u128 << 59) as f64,
+    (1_u128 << 60) as f64,
+    (1_u128 << 61) as f64,
+    (1_u128 << 62) as f64,
+    (1_u128 << 63) as f64,
+    (1_u128 << 64) as f64,
+    (1_u128 << 65) as f64,
+    (1_u128 << 66) as f64,
+    (1_u128 << 67) as f64,
+    (1_u128 << 68) as f64,
+    (1_u128 << 69) as f64,
+    (1_u128 << 70) as f64,
+    (1_u128 << 71) as f64,
+    (1_u128 << 72) as f64,
+    (1_u128 << 73) as f64,
+    (1_u128 << 74) as f64,
+    (1_u128 << 75) as f64,
+    (1_u128 << 76) as f64,
+    (1_u128 << 77) as f64,
+    (1_u128 << 78) as f64,
+    (1_u128 << 79) as f64,
+    (1_u128 << 80) as f64,
+    (1_u128 << 81) as f64,
+    (1_u128 << 82) as f64,
+    (1_u128 << 83) as f64,
+    (1_u128 << 84) as f64,
+    (1_u128 << 85) as f64,
+    (1_u128 << 86) as f64,
+    (1_u128 << 87) as f64,
+    (1_u128 << 88) as f64,
+    (1_u128 << 89) as f64,
+    (1_u128 << 90) as f64,
+    (1_u128 << 91) as f64,
+    (1_u128 << 92) as f64,
+    (1_u128 << 93) as f64,
+    (1_u128 << 94) as f64,
+    (1_u128 << 95) as f64,
+    (1_u128 << 96) as f64,
+    (1_u128 << 97) as f64,
+    (1_u128 << 98) as f64,
+    (1_u128 << 99) as f64,
+    (1_u128 << 100) as f64,
+    (1_u128 << 101) as f64,
+    (1_u128 << 102) as f64,
+    (1_u128 << 103) as f64,
+    (1_u128 << 104) as f64,
+    (1_u128 << 105) as f64,
+    (1_u128 << 106) as f64,
+    (1_u128 << 107) as f64,
+    (1_u128 << 108) as f64,
+    (1_u128 << 109) as f64,
+    (1_u128 << 110) as f64,
+    (1_u128 << 111) as f64,
+    (1_u128 << 112) as f64,
+    (1_u128 << 113) as f64,
+    (1_u128 << 114) as f64,
+    (1_u128 << 115) as f64,
+    (1_u128 << 116) as f64,
+    (1_u128 << 117) as f64,
+    (1_u128 << 118) as f64,
+    (1_u128 << 119) as f64,
+    (1_u128 << 120) as f64,
+    (1_u128 << 121) as f64,
+    (1_u128 << 122) as f64,
+    (1_u128 << 123) as f64,
+    (1_u128 << 124) as f64,
+    (1_u128 << 125) as f64,
+    (1_u128 << 126) as f64,
+    (1_u128 << 127) as f64,
+];
+
+pub const D_SHIFT_F32: [f32; 128] = [
```

Comment:
These should definitely be done using Rust macros.

Reply:
Be my guest